src/build.cc

```diff
@@ -508,21 +518,27 @@ vector<Edge*> RealCommandRunner::GetActiveEdges() {
 
 void RealCommandRunner::Abort() {
   subprocs_.Clear();
+  if (tokens_)
+    tokens_->Clear();
 }
 
 bool RealCommandRunner::CanRunMore() {
   size_t subproc_number =
       subprocs_.running_.size() + subprocs_.finished_.size();
   return (int)subproc_number < config_.parallelism
-    && ((subprocs_.running_.empty() || config_.max_load_average <= 0.0f)
-        || GetLoadAverage() < config_.max_load_average);
+    && (subprocs_.running_.empty() ||
+        ((config_.max_load_average <= 0.0f ||
+          GetLoadAverage() < config_.max_load_average)
+      && (!tokens_ || tokens_->Acquire())));
```

Comment:
I don't think this is correct. The configured maximum number of jobs should be ignored when the jobserver is used. With the current code, the number of jobs is always limited to GuessParallelism() even if there are tokens available.

Reply:
Fixed in the latest patch set

Reply:
@michaelolbrich actually back then this was *intentional* and made sense with my proposed change to the [AOSP build system](https://android-review.googlesource.com/231561) as it always passes `-jN` down to ninja.

I'm not sure any more why I didn't realize that this isn't the correct behaviour for `+ninja` when running tests on a non-Android build system. And since then I forgot this little detail...